### ISSUE TITLE
Some questions about TAM and DCM

### ISSUE THREAD

**Issue by LanariLorenzo** [uta-smile/TVT#2]
Hi, thank you very much for publishing the code I've found your paper really interesting and was trying to figure if this can be adapted to my own model (I'm using basically the same configuration as yours).
The paper highlights two main components DCM and TAMand i was wondering if this two modules can be extracted from the code. 

1. Looking at the code I get that DCM is just a classification head which is performing a classification task on the domain, as DANN (I also see GRL). Am I correct?

2. I am having a harder life finding what is called TAM. Is it something that is built inside the blocks? Because I see that the main change is that the adversarial net is used inside each block to generate a loss_ad . Is this what you call patch-level discriminator inside the paper? Is there need to add something else to create the so called TAM

Thank you very much again for sharing the code

**Comment by viyjy** [comment]
Hi, thanks for your interests in our paper.
1. No, DCM is used to minimize the entropy of the target logits. DCM is totally different from DANN, since there is no adversarial learning in DCM. To apply DCM to your model, you can simply use `lossZoo.im(target_logits)`
2. Right, you are correct.

**Comment by LanariLorenzo** [comment]
Thank you very much for the answer. 
I have one last question, I do see that a global adversarial loss is computed:
         
loss_ad_global = lossZoo.adv(torch.cat((x_s[:,0], x_t[:,0]), 0), ad_net)

this is a plain adversarial loss computed as DANN, correct?

**Comment by viyjy** [comment]
Right.
